hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/AuthenticationBaseMessageTask.java

```diff
@@ -78,12 +78,14 @@ protected void setEndpoint() {
 
     private void checkExistingEndpoint() {
         if (null != principal) {
-            endpoint = endpointManager.getEndpoint(principal.getUuid());
+            clientUuid = principal.getUuid();
+            endpoint = endpointManager.getEndpoint(clientUuid);
             if (null != endpoint) {
                 Connection previousConnection = endpoint.getConnection();
                 if (null != previousConnection && !connection.equals(previousConnection)) {
                     previousConnection.close("A new authentication request from the same client with uuid " + clientUuid
-                            + " is received. Closing the existing connection for this endpoint.", null);
+                            + " is received. Closing the existing connection " + previousConnection + " for this endpoint.",
```

Comment:
Small improvement. If you talk about existing connection in the comment, but previous connection in the java code; it becomes a bit confusing.